internal/ostree/ostree.go

```diff
@@ -50,13 +58,57 @@ func VerifyRef(ref string) bool {
 // ResolveRef resolves the URL path specified by the location and ref
 // (location+"refs/heads/"+ref) and returns the commit ID for the named ref. If
 // there is an error, it will be of type ResolveRefError.
-func ResolveRef(location, ref string) (string, error) {
+func ResolveRef(location, ref string, consumerCerts bool, s *rhsm.Subscriptions) (string, error) {
 	u, err := url.Parse(location)
 	if err != nil {
 		return "", NewResolveRefError(err.Error())
 	}
 	u.Path = path.Join(u.Path, "refs/heads/", ref)
-	resp, err := http.Get(u.String())
+
+	var client *http.Client
+	if consumerCerts {
+		subs := s
```

Comment:
no reason at all

Reply:
Is it written to?  Either way, this wont copy the value, just the pointer, so it's not really doing anything.

Reply:
right but then if i write it straight into s*; the function will have a weird side-effect if s* is nil, no?

Reply:
Ah no I'm wrong here, yea this makes no sense.

Reply:
Oh.  I see it now.  I think it would be the same either way (as in, even if we used `s` everywhere, it wouldn't be overwritten by `subs, err = rhsm.LoadSystemSubscriptions()`, but I think it's better this way, yeah.

Keep it.